openquake/engine/aristotle.py

```diff
@@ -54,32 +55,33 @@ class AristotleParam:
     ignore_shakemap: bool = False
 
 
-def get_close_mosaic_models(lon, lat, max_dist):
+def get_close_mosaic_models(lon, lat, buffer_radius):
     """
     :param lon: longitude
     :param lat: latitude
-    :param max_dist: max dist with respect to a model to consider it relevant
-    :returns: list of mosaic models with the max_dist distance
+    :param buffer_radius: radius of the buffer around the point.
+        This distance is in the same units as the point's
+        coordinates (i.e. degrees), and it defines how far from
+        the point the buffer should extend in all directions,
+        creating a circular buffer region around the point
+    :returns: list of mosaic models intersecting the circle
+        centered on the given coordinates having the specified radius
     """
     mosaic_df = readinput.read_mosaic_df(buffer=1)
-    hypocenter = geo.Point(lon, lat)
-    minx, miny, maxx, maxy = geo.utils.get_bounding_box([hypocenter], max_dist)
-    close_mosaic_models = set()
-    bbox_vertices = [(minx, maxy), (maxx, maxy), (maxx, miny), (minx, miny)]
-    for vertex in bbox_vertices:
-        lonlats = numpy.array([vertex[0], vertex[1]])
-        [mosaic_model] = geo.utils.geolocate([lonlats], mosaic_df)
-        close_mosaic_models.add(mosaic_model)
-    close_mosaic_models = sorted([model for model in close_mosaic_models
+    hypocenter = Point(lon, lat)
+    hypo_buffer = hypocenter.buffer(buffer_radius)
+    geoms = numpy.array([hypo_buffer])
+    close_mosaic_models = geo.utils.geolocate_geometries(geoms, mosaic_df)
+    close_mosaic_models = sorted([model for model in close_mosaic_models[0]
```

Comment:
Why `close_mosaic_models` does not simply the codes different from '???', so that there is no need to filter after?

Reply:
Done